docs/sync-rules/ONEDRIVE.md

```diff
@@ -0,0 +1,57 @@
+### Setting up the OneDrive connector
+
+See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.
+
+### Overview
+
+The advanced sync rules consists of a set of rules where each rule can have one or more parameters out of these:
+
+1. userMailAccounts: list of user emails for which the rule would execute.
+2. skipFilesWithExtensions: list of file extensions to be skipped for the sync.
+3. parentPathPattern: glob pattern specifying the path where the files to be synced are present. 
```

Comment:
You are right. Both  `skipFilesWithExtensions`, `parentPathPattern` does not work on query time, instead they work after the data is fetched from the source. Dana suggested to keep them since it would prevent querying of actual content of these files filtered from these parameters. So in a way, metadata is fetched but the content is not queried.